### ISSUE TITLE
Versions and compatibility issues

### ISSUE THREAD

**Issue by MestreLion** [pop-os/repolib#63]
TL;DR: Are you willing to accept a few PRs to improve packaging and compatibility?

First of all, congrats for such an amazing library! It's been a while I've wanted for an `apt-add-repository` that did the right thing regarding signing keys: to store them in `/etc/apt/keyrings` and add the corresponding `Signed-By` entry to the deb line/stanza. Turns out `Repolib` does that and much, much more! Huge thanks!

So, I've noticed the version in PopOS PPA for Ubuntu Focal is very outdated, and tried to make current 2.1.1 works there. So I started hacking and researching the dependencies and versions to map all incompatibilities, and as it turns out I've managed not only to get 2.1.1 working on Focal but also on Bionic! Yes, using the ancient python 3.6!

The main incompatibilities to current `master` are:
- [`httplib2.error`](https://github.com/pop-os/repolib/blob/master/repolib/command/add.py#L23) was added in `httplib2 0.19.0`, released in 2021-02-07, so not available to Focal.
- [`Path.unlink(missing_ok)`](https://github.com/pop-os/repolib/blob/master/repolib/file.py#L463-L465) requires Python 3.8, available to Bionic but not the default (so `debian/control` should require it)
- [`distro.codename()`](https://github.com/pop-os/repolib/blob/master/repolib/util.py#L55-L56)'s semantic changed in `distro 1.4.0`, released in 2019-02-04, so cannot be used as an `lsb_release` drop-in replacement in Bionic.

While researching all this, I've also noticed many packages and dependencies are not properly configured in `debian/control`, either not listed at all (such as `python3-distro`) or missing/wrong minimum version requirement. I was also able to fix a few lintian warnings and errors.

That said, how do you feel about a few PRs to improve DEB and Python packaging, and possibly make current releases compatible with Focal (and maybe even Bionic)?

**Comment by jacobgkau** [comment]
Thanks for checking in! We've had a few people in the Pop!_OS Mattermost chat have issues due to missing minimum version requirements for some dependencies. PRs to improve that kind of thing would be greatly appreciated.

For compatibility with older versions, we have a dedicated branch for Bionic (https://github.com/pop-os/repolib/tree/master_bionic) that is no longer being updated; the main priority for development is always the current version of Pop!_OS (and the current LTS, which is currently the same version.) However, if you can improve compatibility with older/other distros while not negatively affecting functionality on Pop!_OS 22.04, then we would typically accept the contribution (subject to engineering review, so it's best if it doesn't drastically increase the complexity of the code, too.)

**Comment by jacobgkau** [comment]
Was this addressed in https://github.com/pop-os/repolib/pull/62, https://github.com/pop-os/repolib/pull/71, and/or any of the other PRs since this issue was opened?

**Comment by MestreLion** [comment]
@jacobgkau **yes**, at least concerning dependencies in `debian/control` and such. As for Bionic/Focal compatibility, I didn't have the time to create the PR's (it's working on my system now with local changes), Bionic has EOL'd and soon I'll upgrade everything to 24.04, so this will not be an issue anymore, so I'd say **yes** too.
